### PR TITLE
Geant4 fast sim: tklayout

### PR DESCRIPTION
**Pull request by zaborowska**
New model for tracker smearing introduced, the resolutions obtained using [tkLayout](http://tklayout.web.cern.ch/) and read in FCCSW.
Documentation updated accordingly.
Few tests added.

### CODE REVIEW

**Comment by hegner** [comment]
test this please

**Comment by zaborowska** [comment]
I changed the input Pythia file to the one Valentin generated (and that is now on afs), including decays only within the beam pipe (inside the world volume).
Now all tests (should) pass.

**Comment by jlingema** [comment]
retest this please

**Comment by jlingema** [comment]
retest this please

File: Sim/SimG4Fast/src/components/G4ParticleSmearFormula.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+StatusCode G4ParticleSmearFormula::finalize() {
+  return GaudiTool::finalize();
+}
+
+StatusCode G4ParticleSmearFormula::smearMomentum( CLHEP::Hep3Vector& aMom, int /*aPdg*/) {
+  if(!m_resolutionMomentum->IsValid()) {
+      error()<<"Unable to smear particle's momentum - no resolution given!"<<endmsg;
+      return StatusCode::FAILURE;
+    }
+  m_gauss.initialize(m_randSvc, Rndm::Gauss(1,m_resolutionMomentum->Eval(aMom.mag())));
```
Comment (jlingema) [inline Sim/SimG4Fast/src/components/G4ParticleSmearFormula.cpp]: if I understand [the doc of Rndm::Numbers](http://proj-gaudi.web.cern.ch/proj-gaudi/releases/v26r2/doxygen/d5/dbc/class_rndm_1_1_numbers.html#a151f5a57f09315efacf02887b24e22c1) correctly, I believe this won't work as intended.

File: Sim/SimG4Components/src/G4SaveSmearedParticles.cpp [hunk 2]
```diff
@@ -29,37 +29,42 @@ G4SaveSmearedParticles::G4SaveSmearedParticles(const std::string& aType, const s
...
-      core.Status = 1; // how it is defined ???? as in HepMC ?
-      core.Charge = g4particle->GetCharge();
-      core.P4.Px = info->endMomentum().x()*sim::g42edm::energy;
-      core.P4.Py = info->endMomentum().y()*sim::g42edm::energy;
-      core.P4.Pz = info->endMomentum().z()*sim::g42edm::energy;
-      core.P4.Mass = g4particle->GetMass()*sim::g42edm::energy;
-      core.Vertex.X = info->vertexPosition().x()*sim::g42edm::length;
-      core.Vertex.Y = info->vertexPosition().y()*sim::g42edm::length;
-      core.Vertex.Z = info->vertexPosition().z()*sim::g42edm::length;
+      sim::ParticleInformation* inf = dynamic_cast<sim::ParticleInformation*>(g4particle->GetUserInformation());
```
Comment (jlingema) [inline Sim/SimG4Components/src/G4SaveSmearedParticles.cpp]: rather call this `info`?

File: Sim/SimG4Fast/src/components/G4ParticleSmearTklayout.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  debug()<<"Reading from resolutions file: "<<f.GetName()<<endmsg;
+  while(lnk) {
+    if(sscanf(lnk->GetObject()->GetName(), "etafrom%lf_etato%lf", &lowEta, &highEta)==2) {
+      m_momentumResolutions[highEta] = std::unique_ptr<TGraph>((TGraph*)f.Get(lnk->GetObject()->GetName())->Clone());
+    }
+    lnk=lnk->Next();
+  }
+  m_maxEta = (--m_momentumResolutions.end())->first;
+  debug()<<"maximum pseudorapidity value of the read resolutions: "<<m_maxEta<<endmsg;
+  // TODO here sort map in eta to make sure it is increasing
```
Comment (jlingema) [inline Sim/SimG4Fast/src/components/G4ParticleSmearTklayout.cpp]: file should be closed at the end of this method.

File: Sim/SimG4Fast/src/components/G4ParticleSmearFormula.cpp (hunk 1 above)
Comment (zaborowska) [inline Sim/SimG4Fast/src/components/G4ParticleSmearFormula.cpp]: True.... 
I added Rndm::Numbers::finalize(). Should be enough I guess. Unless you have some other idea how to shoot just one number from the distribution?

**Comment by jlingema** [comment]
retest this please

**Comment by jlingema** [comment]
@hegner, what is our approach for binary files (I just spotted that in this PR a ROOT-file is included)? Do you agree to put them on afs for now until we have set-up a "data repository"?

**Comment by hegner** [comment]
@jlingema - Yes, put them onto AFS/CVMFS for now.

**Comment by jlingema** [comment]
OK, I put the file in `/afs/cern.ch/exp/fcc/sw/0.7/testsamples`.

**Comment by jlingema** [comment]
retest this please

**Comment by jlingema** [comment]
retest this please

**Pull request merged**